File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
  def evalRPN(self, tokens: List[str]) -> int:
    st = list()
    st.append(tokens[0])
    for ch in tokens[1:]:
      if ch == '*':
        a = int(st.pop())
        b = int(st.pop())
        st.append(str(a*b))
      elif ch == '/':
        b = int(st.pop())
        a = int(st.pop())
        st.append(str(int(a/b)))
      elif ch == '+':
        a = int(st.pop())
        b = int(st.pop())
        st.append(str(a+b))
      elif ch == '-':
        b = int(st.pop())
        a = int(st.pop())
        st.append(str(a-b))
      else:
        st.append(ch)
    return int(st.pop())

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
      stack<string>st;
      for(string ch:tokens){
         if(ch == "*"){
            long long a = stol(st.top());
            st.pop();
            long long b = stol(st.top());
            st.pop();
            st.push(to_string(a*b));
          }
          else if(ch ==  "/"){
            long long b = stol(st.top());
            st.pop();
            long long a = stol(st.top());
            st.pop();
            st.push(to_string(a/b));
          }
          else if(ch == "-"){
            long long b = stol(st.top());
            st.pop();
            long long a = stol(st.top());
            st.pop();
            st.push(to_string(a-b));
          }
          else if(ch == "+"){
            long long a = stol(st.top());
            st.pop();
            long long b = stol(st.top());
            st.pop();
            st.push(to_string(a+b));
          }
          else{
            st.push(ch);
          }
        }
        return stoi(st.top());
      }
};